crates/api_models/src/payments.rs

```diff
@@ -293,6 +293,121 @@ pub struct PaymentsGetIntentRequest {
     pub id: id_type::GlobalPaymentId,
 }
 
+#[derive(Debug, serde::Serialize, serde::Deserialize, Clone, ToSchema)]
+#[serde(deny_unknown_fields)]
+#[cfg(feature = "v2")]
+pub struct PaymentsUpdateIntentRequest {
+    pub amount_details: Option<AmountDetails>,
+
+    /// Unique identifier for the payment. This ensures idempotency for multiple payments
+    /// that have been done by a single merchant.
+    #[schema(
+        value_type = Option<String>,
+        min_length = 30,
+        max_length = 30,
+        example = "pay_mbabizu24mvu3mela5njyhpit4"
+    )]
+    pub merchant_reference_id: Option<id_type::PaymentReferenceId>,
+
+    /// The routing algorithm id to be used for the payment
+    #[schema(value_type = Option<String>)]
+    pub routing_algorithm_id: Option<id_type::RoutingId>,
+
+    #[schema(value_type = Option<CaptureMethod>, example = "automatic")]
```

Comment:
Should the `value_type` be omitted for fields having same value as `value_type`

Reply:
No, we are specifying the value type here because the enum is type qualified. Type qualified usages do not work with ToSchema macro